### ISSUE TITLE
regression in gdbstub breakpoints

### ISSUE THREAD

**Issue by comex** [yuzu-emu/yuzu#8503]
This is happening to me on b321c3937137120f27a54ae01decf94f7fa781d1 but it didn't happen on 0d5792cc57112c7db54cbd362d1dcdadae6ec27d from a few days ago.

Breakpoints (as in standard ones, not watchpoints) now don't work at all with my copy of GDB.

It appears to be because the stop response now reports the PC as being 4 bytes after the breakpoint address, rather than the breakpoint address itself.  For example, after setting a breakpoint on 0x88bcd30 and triggering it, I get:
```
T0520:34CD8B0800000000;1f:60BB505810000000;1e:442E610800000000;thread:13;
      ^^^^^^^^ bad PC value
```
This causes gdb to print:
```
Thread 2 "Prepare Thread" received signal SIGTRAP, Trace/breakpoint trap.
```
instead of printing that it stopped on the breakpoint.  And after continuing, the game crashes; I didn't investigate why.

**Comment by comex** [comment]
I'm guessing it's related to this (cc @liamwhite):
```diff
     void ExceptionRaised(u32 pc, Dynarmic::A32::Exception exception) override {
-        if (parent.system.DebuggerEnabled()) {
-            parent.jit.load()->Regs()[15] = pc;
+        if (debugger_enabled) {
+            parent.SaveContext(parent.breakpoint_context);
```

**Comment by liamwhite** [comment]
Fixed in #8490, pull that until it's merged

**Comment by comex** [comment]
Okay.  Off-topic, but I just wanted to express my huge appreciation that you're resurrecting the GDB stub.  I've already gotten significant use out of it in the last week, and I expect to get more in the future, especially since you implemented watchpoints.  Thanks! :)
